src/Raven.Server/Documents/Indexes/MapReduce/ReduceKeyProcessor.cs

```diff
@@ -123,15 +126,15 @@ public void Process(object value)
 
             if (value is decimal)
             {
-                var l = (decimal)value;
+                var d = (decimal)value;
 
                 switch (_mode)
                 {
                     case Mode.SingleValue:
-                        _singleValueHash = Hashing.XXHash64.Calculate((byte*)&l, sizeof(decimal));
+                        _singleValueHash = Hashing.XXHash64.Calculate((byte*)&d, sizeof(decimal));
```

Comment:
You can just cast it to ulong as well. It is 8 bytes, so it just will use the same bit pattern